### DISCUSSION TITLE
「ルーティング」の質問

### DISCUSSION THREAD

**Discussion by tokuda109** [vuejs-jp/learn.nuxt.com#78] [ハンズオン Q&A] [answered]
https://github.com/vuejs-jp/learn.nuxt.com/blob/main/content/2.concepts/3.routing/index.md で何か困ったことや質問があれば遠慮なく質問して下さい

**Comment by kyagi-0826** [comment]
「index.vue」だけどURLは「/Login」などにすることはできたりするのでしょうか

**Reply by k-miyake** [reply, marked as answer]
ご質問ありがとうございます。

Nuxt では、チュートリアルに記載のファイルベースルーティングによる.URL の設定以外に、`nuxt.config.ts` 内でルートを任意の文字列に変更することができるカスタムルーティングの機能があります。
以下のドキュメントに詳細が記載されていますので参考にしてみてください。
https://nuxt.com/docs/guide/recipes/custom-routing

**Reply by kyagi-0826** [reply]
ありがとうございます。見てみます

**Comment by takyafumin** [comment]
`pages/posts/[id].vue`にて、`id`に対して`route.params.id`を直接設定せず、`computed()`を使っているかと思います。
`computed()`を使わないとリアクティブにならない、ということかと思いますが、URLが変わる場合にリアクティブでないと困るケースはどのような動作がありますでしょうか？

**Reply by ubugeeei** [reply]
例えば，自分が知ってるものだと，値の読み取りを created lifecycle で固定してしまった場合 (setup の root で `const id = route.params.id` のように参照を外してしまった場合)，KeepAlive 等でページをキャッシュしてしまった際に古い id が残ってしまうケースはあったりします．
(※ `route.params` 自体は reactive です)

他にもこんなケースもあるって方知ってたらぜひコメントしてほしい 👀

**Reply by takyafumin** [reply]
ありがとうございます！
